docs/dev/experimental-mode.adoc

```diff
@@ -19,74 +19,104 @@ Experimental mode can be enabled by setting odo preferences as well.
+
 ```
+$ odo catalog list components
+Odo OpenShift Components:
+NAME                       PROJECT       TAGS                                        SUPPORTED
+nodejs                     openshift     10,8,8-RHOAR,latest                         YES
+codewind-odo-openjdk18     che           latest                                      YES
+dotnet                     openshift     2.0,latest                                  NO
+httpd                      openshift     2.4,latest                                  NO
+nginx                      openshift     1.10,1.12,1.8,latest                        NO
+nodejs                     openshift     6                                           NO
+perl                       openshift     5.24,5.26,latest                            NO
+php                        openshift     7.0,7.1,latest                              NO
+python                     openshift     2.7,3.5,3.6,latest                          NO
+ruby                       openshift     2.3,2.4,2.5,latest                          NO
+wildfly                    openshift     10.0,10.1,11.0,12.0,13.0,8.1,9.0,latest     NO
 
-If experimental mode is not enabled, `--devfile` flag is not seen in the `odo help push`
-```sh
-$ odo push -h
-Push source code to a component.
-
-Usage:
-  odo push [component name] [flags]
-
-Examples:
-  # Push source code to the current component
-  odo push
-  
-  # Push data to the current component from the original source.
-  odo push
-  
-  # Push source code in ~/mycode to component called my-component
-  odo push my-component --context ~/mycode
-
-Flags:
-      --config               Use config flag to only apply config on to cluster
-      --context string       Use given context directory as a source for component settings
-  -f, --force-build          Use force-build flag to force building the component
-  -h, --help                 Help for push
-      --ignore stringSlice   Files or folders to be ignored via glob expressions.
-      --show-log             If enabled, logs will be shown when built
-      --source               Use source flag to only push latest source on to cluster
-
-Additional Flags:
-  -v, --v Level              Log level for V logs. Level varies from 0 to 9 (default 0).
-      --vmodule moduleSpec   Comma-separated list of pattern=N settings for file-filtered logging
+Odo Devfile Components:
+NAME                 DESCRIPTION                           SUPPORTED
+maven                Upstream Maven and OpenJDK 11         YES
+nodejs               Stack with NodeJS 10                  YES
+openLiberty          Open Liberty microservice in Java     YES
+java-spring-boot     Spring Boot® using IBM Java           YES
 ```
 
-If the experimental mode is disabled and the hidden flag is used, it will throw an unknown flag error.
-```sh
-$ odo push --devfile ./devfile.yaml  
-Error: unknown flag: --devfile
+##### Creating a devfile component:
```

Comment:
IMO, this section should mention that `odo create` will create a `devfile.yaml` in `pwd`.